chain/chain/src/chain.rs

```diff
@@ -1847,7 +1848,8 @@ impl Chain {
     pub fn get_final_transaction_result(
         &mut self,
         hash: &CryptoHash,
-    ) -> Result<FinalExecutionOutcomeView, Error> {
+        fetch_receipt: bool,
+    ) -> Result<FinalExecutionOutcomeViewEnum, Error> {
```

Comment:
As far as I can tell, *View structs were designed to only be constructed by the view client to match the internal representations to semi-public interfaces. Thus, I am surprised to see View constructed in chain.rs.

Reply:
Agree but I think it is outside the scope of this PR